wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt

```diff
@@ -173,14 +155,43 @@ class MainViewModel @Inject constructor(
         }
     }
 
+    private fun updateEntityStates(entity: Entity<*>) {
+        if (supportedDomains().contains(entity.domain)) {
+            entities[entity.entityId] = entity
+            // add to cache if part of favorites
+            if (favoriteEntityIds.value.contains(entity.entityId)) {
+                addCachedFavorite(entity.entityId)
+            }
+        }
+    }
+
+    suspend fun updateUI() = withContext(Dispatchers.IO) {
+        val getAreaRegistry = async { homePresenter.getAreaRegistry() }
+        val getDeviceRegistry = async { homePresenter.getDeviceRegistry() }
+        val getEntityRegistry = async { homePresenter.getEntityRegistry() }
+        val getEntities = async { homePresenter.getEntities() }
+
+        areaRegistry = getAreaRegistry.await()?.also {
+            areas.clear()
+            areas.addAll(it)
+        }
+        deviceRegistry = getDeviceRegistry.await()
+        entityRegistry = getEntityRegistry.await()
+
+        entities.clear()
+        getEntities.await()?.forEach {
+            updateEntityStates(it)
+        }
```

Comment:
Theoretically, the entities might not have been loaded, so it should only be cleared after `await()`

```suggestion
        getEntities.await()?.also {
            entities.clear()
            it.forEach { state -> updateEntityStates(state) }
        }
```